File: solutions/opus-1/opus/python/transpose.py
import sys

def main():
    data = sys.stdin.buffer.read().split()
    if len(data) < 2:
        return
    r, c = int(data[0]), int(data[1])
    vals = data[2:2 + r * c]
    out = []
    for j in range(c):
        out.append(b" ".join(vals[i * c + j] for i in range(r)))
    sys.stdout.buffer.write(b"\n".join(out) + (b"\n" if out else b""))

main()

File: solutions/opus-1/opus/ruby/transpose.rb
tokens = $stdin.read.split
r = tokens[0].to_i
c = tokens[1].to_i
vals = tokens[2, r * c] || []
out = []
c.times do |j|
  out << (0...r).map { |i| vals[i * c + j] }.join(" ")
end
print out.map { |l| l + "\n" }.join
